runners/flink/runner/src/main/java/org/apache/beam/runners/flink/TestFlinkRunner.java

```diff
@@ -79,6 +86,31 @@ public PipelineResult run(Pipeline pipeline) {
   public PipelineOptions getPipelineOptions() {
     return delegate.getPipelineOptions();
   }
+
+  private void assertAssertionCounters(Pipeline pipeline, FlinkRunnerResult result) {
+    int expectedNumberOfAssertions = PAssert.countAsserts(pipeline);
+    Integer succeededAssertions = 0;
+    try {
+      succeededAssertions = result.getAggregatorValue(PAssert.SUCCESS_COUNTER);
+    } catch (AggregatorRetrievalException e) {
+      // No assertions registered will cause an AggregatorRetrievalException here.
+    }
+    Integer failedAssertions = 0;
+    try {
+      failedAssertions = result.getAggregatorValue(PAssert.FAILURE_COUNTER);
+    } catch (AggregatorRetrievalException e) {
+      // No assertions registered will cause an AggregatorRetrievalException here.
+    }
+    assertThat(
```

Comment:
The documentation on `TestPipeline` says that you have to throw the failing assertion message (https://github.com/apache/beam/blob/master/sdks/java/core/src/main/java/org/apache/beam/sdk/testing/TestPipeline.java#L95)